nexus/src/nexus.rs

```diff
@@ -2249,6 +2259,49 @@ impl Nexus {
     pub async fn session_hard_delete(&self, token: String) -> DeleteResult {
         self.db_datastore.session_hard_delete(token).await
     }
+
+    pub async fn updates_refresh_metadata(&self) -> Result<(), Error> {
+        let rack = self.as_rack();
+        let trust_root = self
+            .tuf_trusted_root
+            .as_ref()
+            .ok_or_else(|| Error::InvalidRequest {
+                message: "updates system not configured".into(),
+            })?
+            .clone();
+
+        let artifacts = tokio::task::spawn_blocking(move || {
+            crate::updates::read_artifacts(&rack, &trust_root)
+        })
+        .await
+        // first, the JoinError:
+        .map_err(|e| Error::InternalError { internal_message: e.to_string() })?
```

Comment:
This only happens when the call to `read_artifacts` panics, right?

1. I *think* we're using panic=abort, so this shouldn't happen,
2. ... if it did, panicking ASAP seems like the right behavior.

In other codebases, I just unwrapped the first error for this reason: https://github.com/oxidecomputer/async-bb8-diesel/blob/09f17db081be85abdf4f773a385381637b9c8d76/src/connection.rs#L42-L45